### ISSUE TITLE
createDeprecatedModule should be striped from production builds.

### ISSUE THREAD

**Issue by krisselden** [ember-cli/ember-resolver#132]

**Comment by rwjblue** [comment]
Why? Having the deprecated shims present in dev builds with a deprecation, but not present in prod builds would obviously be a breaking change.

**Comment by rwjblue** [comment]
I could see doing a major bump here to drop them completely though.

**Comment by krisselden** [comment]
This was still in our production build https://github.com/ember-cli/ember-resolver/blob/5c6b080059d27d8533e57db2a5a7e7b24eb93727/vendor/ember-resolver/legacy-shims.js#L5-L9

**Comment by rwjblue** [comment]
Right, that is because there is no infrastructure for auto-stripping these things in general.  https://github.com/ember-cli/rfcs/pull/50 attempts to address this. The `Ember.deprecate` call there will be a noop in production builds of Ember though.

**Comment by stefanpenner** [comment]
@chadhietala this may be good task for you svelete tooling
